auth/services/selfsigned/signer.go

```diff
@@ -0,0 +1,103 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package selfsigned
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	ssi "github.com/nuts-foundation/go-did"
+	"github.com/nuts-foundation/go-did/did"
+	vc2 "github.com/nuts-foundation/go-did/vc"
+	"github.com/nuts-foundation/nuts-node/auth/contract"
+	"github.com/nuts-foundation/nuts-node/auth/services"
+	"github.com/nuts-foundation/nuts-node/vcr/credential"
+	"github.com/nuts-foundation/nuts-node/vcr/signature/proof"
+	"time"
+)
+
+const credentialType = "NutsEmployeeCredential"
+
+func (v sessionStore) SigningSessionStatus(sessionID string) (contract.SigningSessionResult, error) {
+	s, ok := v.sessions[sessionID]
+	if !ok {
+		return nil, services.ErrSessionNotFound
+	}
+	var vp *vc2.VerifiablePresentation
+
+	if s.status == SessionCompleted {
+		issuer := did.MustParseDID(s.Employer) // todo panic << validate in API?
+		expirationData := time.Now().Add(24 * time.Hour)
+		credentialOptions := vc2.VerifiableCredential{
+			Context:           []ssi.URI{credential.NutsV1ContextURI},
+			Type:              []ssi.URI{vc2.VerifiableCredentialTypeV1URI(), ssi.MustParseURI(credentialType)},
+			Issuer:            issuer.URI(),
+			IssuanceDate:      time.Now(),
+			ExpirationDate:    &expirationData,
+			CredentialSubject: s.credentialSubject(),
+		}
+		verifiableCredential, err := v.vcr.Issuer().Issue(context.TODO(), credentialOptions, false, false)
+		if err != nil {
+			return nil, fmt.Errorf("issue VC failed: %w", err)
+		}
+		proofOptions := proof.ProofOptions{
+			Created:      time.Now(),
+			Challenge:    &s.contract,
+			ProofPurpose: "",
+		}
+		vp, err = v.vcr.Holder().BuildVP(context.TODO(), []vc2.VerifiableCredential{*verifiableCredential}, proofOptions, &issuer, true)
+		if err != nil {
+			return nil, fmt.Errorf("build VP failed: %w", err)
+		}
+	}
+
+	return signingSessionResult{
+		id:                     sessionID,
+		status:                 s.status,
+		request:                s.contract,
+		verifiablePresentation: vp,
+	}, nil
+}
+
+func (v sessionStore) StartSigningSession(rawContractText string, params map[string]interface{}) (contract.SessionPointer, error) {
+	sessionBytes := make([]byte, 16)
+	rand.Reader.Read(sessionBytes)
+
+	sessionID := hex.EncodeToString(sessionBytes)
+	s := session{
+		contract: rawContractText,
+		status:   SessionCreated,
+	}
+	// load params directly into session
+	marshalled, err := json.Marshal(params)
+	// only functions or other weird constructions can cause an error here. No need for custom error handling.
+	if err != nil {
+		return nil, err
+	}
+	// impossible to get an error here since both the pointer and the data is under our control.
+	_ = json.Unmarshal(marshalled, &s)
+	v.sessions[sessionID] = s
```

Comment:
should we check for existence of sessionID?

Reply:
It's a 16 byte secure rand with a several minute lifetime